src/visitors/mod.rs

```diff
@@ -1,2 +1,2 @@
-pub mod ast;
-pub mod semantic;
+pub(crate) mod ast;
+pub(crate) mod semantic;
```

Comment:
I'm not familiar with this syntax. Could you explain what this does?

Reply:
It makes the visibility public to the crate, but not publicly accessible. We essentially had this already, since the parent package is not public, but I made this explicit here.